src/testModel.ts

```diff
@@ -383,6 +386,17 @@ export class TestModel extends DisposableBase {
     return this._filesPendingListTests.promise;
   }
 
+  private _closeTraceViewerIfNeeded() {
+    if (
+      this._collection.selectedModel() !== this ||
+      // check if there's a high priority trace viewer that supports currennt configuration
+      this._traceViewers.find(t => t.checkSupport(false)) !== this._currentTraceViewer
+    ) {
+      this._currentTraceViewer?.close();
```

Comment:
The name does not make sense to me - I think once you closed the trace viewer you can't use it. So either traceViewer should be a traceViewerFactory, or this should be closeTraceView().

Reply:
I was keeping the same pattern from `TraceViewer` before my changes: one instance with the same lifecycle as its container (before it was `extension`, now it's `testModel`) that could be closed and open several times.

And the idea of this `_closeTraceViewerIfNeeded` is to only close the trace viewer when either the selected model actually changes or when current trace viewer becomes due to settings changes (e.g., checking the embedded setting would close the spawn trace viewer and open the embedded),

There are situations where changing trace related settings won't close the current trace viewer. For instance, if it currently has embedded checked but playwright version doesn't support it, it will open spawn trace viewer. If the user then unchecks the embedded setting, it should keep the spawn trace viewer open.

Reply:
Let's land this bit as is and if I come up with a better name I'll follow up.

Reply:
I left the `_closeTraceViewerIfNeeded` function but created a traceViewerFactory in 31a15f3f926c5f17a8e16a4476d52c62f11778a9 to check which trace viewer strategy to instanciate.